python_modules/dagster/dagster/core/definitions/pipeline_base.py

```diff
@@ -61,7 +71,12 @@ def _resolve_solid_selection(self, solid_selection):
             )
         return solids_to_execute
 
-    def _subset_for_execution(self, solids_to_execute, solid_selection=None):
+    def _subset_for_execution(self, solids_to_execute, solid_selection=None, asset_selection=None):
+        if asset_selection:
+            return InMemoryPipeline(
+                self._pipeline_def.get_job_def_for_asset_selection(asset_selection),
+                asset_selection=asset_selection,
+            )
```

Comment:
related to above - it feels like the complexity cost to the code is greater than it ought to be, the subsetting flows between op selection vs asset selection seems like it shouldn't require so much branching and could be more consolidated